src/mmw/apps/modeling/mapshed/calcs.py

```diff
@@ -203,6 +205,38 @@ def manure_spread(aeu):
 
     return n_list, p_list
 
+def ls_factors(lu_strms, areas, avg_slope):
+    results = [0.0] * len(lu_strms)
+    for i in range(len(lu_strms)):
+        results[i] = ls_factor(lu_strms[i] * KM_PER_M, areas[i], avg_slope)
+    return results
+
+def ls_factor(stream_length, area, avg_slope):
+    # units --> stream_length:km, area:hectare, avg_slope:%
+    slope_cell_size = 30 #for posterity
+    if 0 <= avg_slope <= 1.0:
+        m = 0.2
+    elif 1.0 < avg_slope <= 3.5:
+        m = 0.3
+    elif 3.5 < avg_slope <= 4.5:
+        m = 0.4
+    else:
+        m = 0.5
+
+    if slope_cell_size < 50:
+        s = 0.0793
+        t = 0.1913
+    else:
+        s = 1
+        t = 0
+
+    if stream_length < (250.0 * KM_PER_M):
```

Comment:
250 meters. I think the decimal conversion factor convention makes this calculation (even) less clear.